File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.go
func reverseWords(s string) string {
    var j int
    result := ""
    
    for i := 0; i < len(s); i++ {
        if s[i] == ' ' {
            continue
        }
        
        j = i
        for j = i; j < len(s); j++ {
            if s[j] == ' '{
                break
            }
        }
        
        if result == ""{
            result = s[i:j]
        } else{
            result = s[i:j] + " " + result
        }
        i = j
    }
    
    return result
}

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        vector<string> m;
        string word = "";
        for(char a : s) {
            if (a == ' ') {
                if (word.size() != 0) {
                    m.push_back(word);
                    word = "";
                }
            } else {
                word += a;
            }
        }
        
        if (word.size() != 0) {
            m.push_back(word);
            word = "";
        }
        
        string result = "";
        for(int i = m.size() - 1; i > 0; i--){
            result += m[i] + " ";
        }
        result += m[0];
        return result;
    }
};